lib/asset-sprayer/src/prompts.rs

```diff
@@ -0,0 +1,147 @@
+use std::path::PathBuf;
+
+use async_openai::types::{
+    ChatCompletionRequestMessage, ChatCompletionRequestSystemMessageContent,
+    ChatCompletionRequestUserMessageContent, CreateChatCompletionRequest,
+};
+
+use crate::{AssetSprayerError, AssetSprayerResult};
+
+#[derive(Debug, Clone)]
+pub struct Prompts {
+    prompts_dir: Option<PathBuf>,
+}
+
+#[derive(Debug, Clone, Copy, strum::Display)]
+pub enum Prompt {
+    AssetSchema,
+}
+
+impl Prompt {
+    fn yaml_relative_path(&self) -> &str {
+        match self {
+            Prompt::AssetSchema => "aws/asset_schema.yaml",
+        }
+    }
+
+    fn yaml_embedded(&self) -> &'static str {
+        match self {
+            Prompt::AssetSchema => include_str!("../prompts/aws/asset_schema.yaml"),
+        }
+    }
```

Comment:
I get nervous when not using path joining features from the standard library, but I think it's safe to say these path separators in the strings will be fine.